__init__.py

```diff
@@ -476,6 +501,38 @@ def on_pairing_end(self, error_dialog):
         self.state = SetupState.INACTIVE
 
     # Pairing GUI events
+    ### Language Selection & Welcome Screen
+    def handle_welcome_screen(self):
+        self.state = SetupState.WELCOME
+        self.handle_display_manager("Welcome")
+        sleep(0.3) # Wait for display to show the screen before speaking
+        self.speak_dialog("welcome_screen", wait=True)
+        sleep(0.3) # Wait for display a bit before showing the next screen
+        if self.pairing_mode != PairingMode.GUI:
+            self.handle_backend_menu()
+        else:
+            lang_support_enabled = self.settings.get("enable_language_support", False)
+            if not lang_support_enabled:
+                self.handle_backend_menu()
+            else:
+                self.handle_language_menu()
+
+    def handle_language_menu(self):
+        self.state = SetupState.SELECTING_LANGUAGE
+        supported_languages = self.settings["langs"]
+        self.gui["supportedLanguagesModel"] = supported_languages
+        self.handle_display_manager("LanguageMenu")
+        self.speak_dialog("language_menu")
+
+    def handle_language_selected(self, message):
+        self.selected_language = message.data["code"]
+        system_code = message.data["system_code"]
+        self.bus.emit(Message("system.configure.language", data={"code": self.selected_language, "language_code": self.system_code}))
```

Comment:
shouldn't this be system_code instead of self.system_code in the message data?

Reply:
fixed